README.md

```diff
@@ -225,9 +228,175 @@ If the modification of the Application resource is completed successfully, the `
+    ```
+1.  Mount an entry from a `ConfigMap` into the binding `Secret`
+    ```plain
+    “service.binding/certificate”:
+      "path={.status.data.dbConfiguration},objectType=ConfigMap,sourceKey=certificate"
+    ```
+1.  Mount an entry from a `ConfigMap` into the binding `Secret` with a different key
+    ```plain
+    “service.binding/timeout”:
+      “path={.status.data.dbConfiguration},objectType=ConfigMap,sourceKey=db_timeout”
+    ```
+1.  Mount a resource defintion value into the binding `Secret`
+    ```plain
+    “service.binding/uri”:
+      "path={.status.data.url}"
+    ```
+1.  Mount a resource definition value into the binding `Secret` with a different key
+    ```plain
+    “service.binding/uri":
+      "path={.status.data.connectionURL}”
+    ```
+1.  Mount the entries of a collection into the binding `Secret` selecting the key and value from each entry
+    ```plain
+    “service.binding/endpoints”:
+      "path={.status.bootstrap},elementType=sliceOfMaps,sourceKey=type,sourceValue=url"
+    ```
+
 ## Mapping Existing Values to New Values
 
-Many applications will not be able to consume the secrets exposed by Provisioned Services directly.  Teams creating Provisioned Services do not know how their services will be consumed, teams creating Applications will not know what services will be provided to them, different language families have different idioms for naming and style, and more.  Users must have a way of describing a mapping from existing values to customize the provided entries to ones that are usable directly by their applications.  This specification is described as an extension to the [Service Binding](#service-binding) specification and assumes full compatibility with it.
+Many applications will not be able to consume the secrets exposed by Provisioned Services directly.  Teams creating Provisioned Services do not know how their services will be consumed, teams creating Applications will not know what services will be provided to them, different language families have different idioms for naming and style, and more.  Users should have a way of describing a mapping from existing values to customize the provided entries to ones that are usable directly by their applications.  This specification is described as an extension to the [Service Binding](#service-binding) specification and assumes full compatibility with it.
 
 A Service Binding Resource **MAY** define a `.spec.mappings` which is an array of `Mapping` objects.  A `Mapping` object **MUST** define `name` and `value` entries.  The value of a `Mapping` **MAY** contain zero or more tokens beginning with `((`, ending with `))`, and encapsulating a binding `Secret` key name.  The value of this `Secret` entry **MUST** be substituted into the original `value` string, replacing the token.  Once all tokens have been substituted, the new `value` **MUST** be added to the `Secret` exposed to the resource represented by `application`.
```

Comment:
Can you elaborate a bit on what you'd expect here?  Why wouldn't 

```
mappings:
- name: TEST
  value: "((alpha)), ((bravo)), ((charlie))"
```

be sufficient?

In the other direction (referring to indexed values from an incoming array) is probably something we should add though.  A follow up PR?

Reply:
I'm thinking where the source set isn't know, because it's coming in from somewhere else.. building a redis host / peer list, populated from the current instances of redis that are running.

Going from an array into a formatted list, where you don't know the length of the array in advance, is tricky, especially if the consuming system doesn't enjoy empty list entries, or tolerate trailing commas.. formatting such an array into viable json using `(())` would be challenging.

Reply:
@BarDweller Mapping is scoped to a single secret, and while I guess that secret could have 

```
{
  "node.1": "...",
  "node.2": "...",
  "node.3": "...",
}
```

I wouldn't expect that to be common.  I would expect the Synthetic Provided Service to handle your use-case of enumerating all the Redis host resource in a cluster and pre-building the `endpoints` array into the secret.

This "I cannot predict all of the possible use cases" problem is one of the major reasons I kind of want to remove that extension completely; leave it to platform to decide how sophisticated to make that mapping and leave _this_ mapping a straight-forward 1:1.  Thoughts?

Reply:
Reading other comments, go templates have been proposed to follow later, and I know I can use those to do what I'm looking for here. (Although they are a little awkward/verbose, but they will serve this purpose). So I'm not too worried here now.